tiledb/sm/cpp_api/config.h

```diff
@@ -208,6 +208,16 @@ class Config {
     impl::check_config_error(err);
   }
 
+  /** Compares configs for equality. */
+  bool operator==(const Config& rhs) const {
+    uint8_t equal;
+    tiledb_config_compare(config_.get(), rhs.config_.get(), &equal);
```

Comment:
Check the return value here. If it's not `TILEDB_OK`, we need to throw an exception.